### PR TITLE
Agent pool resource

### PR DESCRIPTION
**Pull request by by-nelson**
### Changes
- Added `DeleteRequest` to the new client.
- Added client Pool Create and Delete methods.
  - Using new client functions.
- Created basic Agent Pool resource.
  - Validating max size greater or equal to 0.
  - Null size sets to unlimited.
- Added acceptance tests to the pool resource.
---
### Tests
The test case can be run with:
```sh
TF_ACC=1 go test -count=1 -run='TestAccPoolResource_basic' -v ./...   
```

### CODE REVIEW

File: teamcity/pool_resource.go [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+// Resource functions implementation
+// returns the full name of the resource
+func (r *poolResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_pool"
+}
+
+// returns the schema of the resource
+func (r *poolResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "An Agent Pool in TeamCity is a group of agents that can be associated to projects. More info [here](https://www.jetbrains.com/help/teamcity/configuring-agent-pools.html)",
```
Comment (ChildOfJustice) [inline teamcity/pool_resource.go:43]: typo: "associated `with` projects"

File: teamcity/pool_resource.go [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var newName string
+	var newSize string
+
+	// verify plan values
+	if plan.Name.IsNull() {
```
Comment (ChildOfJustice) [inline teamcity/pool_resource.go:181]: Here and in the pool_data_source.go#78 too - maybe we could drop this code, wdyt? I suppose `"name": schema.StringAttribute{ Required: true,` is enough

File: teamcity/pool_resource.go [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+	// get values from plan
+	var plan models.PoolDataModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// verify values
+	if plan.Name.IsNull() {
```
[the remark this answers is not in the document]
Comment (by-nelson) [reply teamcity/pool_resource.go:77]: You're right I did this too for the negative size numbers and then I moved it to a schema validator. Let me fix it

**Review by ChildOfJustice — APPROVED** [review]
Thanks! Looks great!

**Comment by ChildOfJustice** [comment]
I found the problem why GitHub cannot merge - we have to rebase (this repo is configured to reject merge commits) and while rebasing your branch onto master, I encountered a very small conflict, now everything is fine, but I had to rebase, so I created another branch and merged it - [agent-pool-resource-rebased](https://github.com/JetBrains/terraform-provider-teamcity/tree/agent-pool-resource-rebased).

Thank you again!

**Pull request closed without merging**